x-pack/plugins/security/public/management/api_keys/api_keys_grid/api_keys_grid_page.tsx

```diff
@@ -589,9 +667,9 @@ export const TypesFilterButton: FunctionComponent<TypesFilterButtonProps> = ({
           hasActiveFilters={query.hasSimpleFieldClause('type', 'managed')}
           onClick={() =>
             onChange(
-              query.hasSimpleFieldClause('type', 'managed')
-                ? query.removeSimpleFieldClauses('type')
-                : query.removeSimpleFieldClauses('type').addSimpleFieldValue('type', 'managed')
+              query.hasClauses()
+                ? query.removeAllClauses()
+                : EuiSearchBar.Query.parse('(metadata.managed:true OR Alerting*)')
```

Comment:
@azasypkin This is the query I'm using to replace the old UI only filter for `metadata.managed = true OR name.indexOf('Alerting: ') === 0`([code on main](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security/public/management/api_keys/api_keys_grid/api_keys_grid_page.tsx#L864))

Reply:
Yeah, supporting two different methods of detecting "managed" API keys looks really messy (one for the fleet with `managed: true` metadata and one for Alerting based on the name), both in terms of UX and technical debt.

@elastic/response-ops, how hard would it be for you to include `managed: true` metadata to the API keys you create so that we can label all "managed" API keys as such in the UI and efficiently query them from ES solely based on the metadata "managed" key convention?

![Screenshot from 2024-03-06 10-46-27](https://github.com/elastic/kibana/assets/1713708/d2aa24d2-5543-4960-933d-2127123f66c4)

Reply:
We create API keys here: https://github.com/elastic/kibana/blob/8be837a15415376d39d7a0182a1995d71ed7888b/x-pack/plugins/alerting/server/rules_client_factory.ts#L140 so seems like it would be pretty straightforward to add `managed:true` to the metadata here. However all the existing API keys for alerting would not have this metadata, unless there is some sort of migration we could do